src/proofnetwork/baserestapi.cpp

```diff
@@ -282,6 +287,58 @@ bool BaseRestApiPrivate::replyShouldBeHandledByError(QNetworkReply *reply) const
     return (errorHundreds != 2 && errorHundreds != 4);
 }
 
+std::tuple<QString, NetworkErrorCode::Code> BaseRestApiPrivate::errorByCheckConnection(QNetworkReply *reply)
+{
+    QHostAddress host;
+    auto result = std::make_tuple(QStringLiteral("Host %1 is unavailable. Try again later").arg(reply->url().host()),
+                                  NetworkErrorCode::Code::ServiceUnavailable);
+
+    if (host.isLoopback() || QNetworkInterface::allAddresses().contains(host))
+        return result;
+
+    bool isIp = host.setAddress(reply->url().host());
+    if (isIp) {
+        const auto &interfaces = QNetworkInterface::allInterfaces();
+        bool atLeastOneInterfaceUp = false;
+        for (const auto &interface : interfaces) {
+            auto flags = interface.flags();
+            if (!(flags & QNetworkInterface::IsLoopBack) && (flags & QNetworkInterface::IsUp)
+                && interface.addressEntries().count() > 0) {
+                atLeastOneInterfaceUp = true;
+                break;
+            }
+        }
+
+        if (!atLeastOneInterfaceUp) {
+            result = std::make_tuple(QStringLiteral("You don't have network connection. Please, connect your device to "
+                                                    "network and try again"),
+                                     NetworkErrorCode::Code::NoNetworkConnection);
+        }
+    } else if (!internetConnectionEstablished()) {
+        result = std::make_tuple(QStringLiteral("Your device seems to be in network without Internet connection. "
+                                                "Please, check if Internet is accessible and try again"),
+                                 NetworkErrorCode::Code::NoInternetConnection);
+    } else if (reply->error() == QNetworkReply::HostNotFoundError) {
+        result = std::make_tuple(QStringLiteral("Host %1 not found. Please check your DNS settings and try again")
+                                     .arg(reply->url().host()),
+                                 NetworkErrorCode::Code::HostNotFound);
+    }
+    return result;
+}
+
+bool BaseRestApiPrivate::internetConnectionEstablished()
```

Comment:
Should have some indication of long process in naming. Maybe tryInternetConnection or something like that